spacelift/data_policies.go

```diff
@@ -0,0 +1,77 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataPolicies() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_policies` can find all policies that have certain labels," +
+			"and return their IDs. ",
+
+		ReadContext: dataPoliciesRead,
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:        schema.TypeString,
+				Description: "type of the policy",
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "required labels to match",
+			},
+			"ids": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataPoliciesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		Policies *structs.Policies `graphql:"policies()"`
+	}
+
+	if err := meta.(*internal.Client).Query(ctx, "PoliciesRead", &query, nil); err != nil {
+		return diag.Errorf("could not query for policy: %v", err)
+	}
+
+	typeRaw, typeSpecified := d.GetOk("type")
+	requestedType := typeRaw.(string)
+	labelsRaw, labelsSpecified := d.GetOk("labels")
+	requestedLabels := labelsRaw.(*schema.Set).List()
+	ids := schema.NewSet(schema.HashString, []interface{}{})
+	for _, policy := range query.Policies.Policies {
+		if typeSpecified && policy.Type != requestedType {
+			continue
+		}
+		if labelsSpecified {
+			for _, required := range requestedLabels {
```

Comment:
I'm not sure about this method to check all requested labels are present. Is there a better way?

Reply:
Looks good.